src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,251 @@
+package edu.berkeley.ce.rockslicing
+
+import scala.annotation.tailrec
+
+/**
+  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
+  * approximately balanced loads among processors.
+  */
+object SeedJointSelector {
+  val THRESHOLD = 0.50
+
+  /**
+    * Searches though sequence of joint sets to find best seed joints
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numProcessors Number of processors to be used in the analysis
+    * @param remainingJoints Seq of Seq of joints that were not selected as seed joints
+    * @return Returns a tuple containing two Seq's of Joints. The first Seq contains the seed joints that
+    *         best divide the input volume. The second Seq contains the joints that remain from the input
+    *         joint set.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numProcessors: Int,
+                      remainingJoints: Seq[Seq[Joint]] = Seq.empty[Seq[Joint]]): (Seq[Joint], Seq[Joint]) = {
+    val allJoints = if (remainingJoints.isEmpty) {
```

Comment:
I have to admit I got confused on reading over this function again. Maybe there's a better name besides `allJoints` for this value. It's also possible you could get rid of this `if` entirely...?

Intuitively, when I'm reading a recursive function, I'm trying to look at how the collections that are passed as arguments change with each call. I can understand how `jointSets` is monotonically decreasing in size since you're calling `tail` with each recursive call, but I don't understand what's happening with `remainingJoints`.

It seems to me that what you'd want to do is store all joint sets that get "passed over" here. So maybe it would be named `unusedJoints` or something like that. The collection starts out empty and, with each recursive call, you add on the joint set that you just skipped over, so your calls would look like `searchSeedJointSets(jointSets.tail, inputVolume, numProcessors, jointSets.head +: remainingJoints`.

Then, once you do find a suitable joint set, you just return `unusedJoints.flatten ++ jointSets.tail.flatten ++ ` whatever joints were not used as seed joints from the joint set that finally worked.

This seems like it would a way to make the recursion clearer, but I'm probably missing something somewhere, or maybe that's what this code is actually doing and I didn't notice :)